public/modules/custom/grants_admin_applications/src/Form/SubmittedApplicationsForm.php

```diff
@@ -207,31 +172,29 @@ public function buildForm(array $form, FormStateInterface $form_state): array {
   /**
    * Resend application callback submit handler.
    */
-  public static function resendApplicationCallback(array $form, FormStateInterface $formState): void {
-    $logger = self::getLoggerChannel();
-    $messenger = \Drupal::service('messenger');
+  public function resendApplicationCallback(array $form, FormStateInterface $formState): void {
     $triggeringElement = $formState->getTriggeringElement();
     try {
 
       $transactionId = $triggeringElement['#id'];
 
       $sParams = [
         'transaction_id' => $transactionId,
-        'lookfor' => 'appenv:' . ApplicationHandler::getAppEnv(),
+        'lookfor' => 'appenv:' . Helpers::getAppEnv(),
       ];
 
-      $res = \Drupal::service('helfi_atv.atv_service')->searchDocuments($sParams);
+      $res = $this->atvService->searchDocuments($sParams);
       $atvDoc = reset($res);
 
       if (!$atvDoc) {
         return;
       }
 
-      self::sendApplicationToIntegrations($atvDoc, $transactionId);
+      $this->sendApplicationToIntegrations($atvDoc, $transactionId);
     }
-    catch (\Exception $e) {
-      $messenger->addError($e->getMessage());
-      $logger->error(
+    catch (GuzzleException | \Exception $e) {
+      $this->messenger()->addError($e->getMessage());
```

Comment:
Printing exception message might expose some sensitive information.

Reply:
And here